media/js/src/main.jsx

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import {getAssets} from './utils';
+
+class Main extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            // Collection tab is default
+            activeTab: 'collection'
+        };
+
+        const me = this;
+        getAssets().then(function(d) {
+            me.setState({
+                assets: d.assets
+            });
+        }, function(e) {
+            me.setState({
+                assets: e
+            });
+        });
+
+        this.clickTab = this.clickTab.bind(this);
+    }
+
+    clickTab(e) {
+        e.preventDefault();
+        const clicked = e.target.text.toLowerCase();
+        this.setState({activeTab: clicked});
+    }
+
+    render() {
+        let assets = [];
+        let assetsDom = 'Loading Assets...';
+
+        if (this.state.assets && typeof this.state.assets === 'string') {
+            // Display error to user
+            assetsDom = <strong>{this.state.assets}</strong>;
+        } else if (this.state.assets) {
+            this.state.assets.forEach(function(asset) {
+                const thumbnail = asset.thumb_url || asset.sources.image.url;
+                assets.push(
+                    <div className="card" key={asset.id}>
```

Comment:
I think this might make sense to break this out into its own stateless component.